styleguide/template.md

```diff
@@ -222,44 +222,34 @@ You can use a [Markdown table generator tool](http://www.tablesgenerator.com/mar
+```markdown
+[!code-<language>[<name>](<pathToFile><queryoption><queryoptionvalue>)]
 ```
 
-The example above shows C# syntax, but other languages are supported.
-Use `code-fsharp` for F# samples; use `code-vbnet` for Visual Basic samples.
-Other languages that are supported are:
-* C++: `code-cpp`
-* HTML: `code-html`
-* JavaScript: `code-javascript`
-* Powershell: `code-ps`
-* SQL: `code-sql`
-* XML: `code-xml`
-
-
+* `-<language>` (*optional* but *recommended*)
+  * Language of the code snippet being referenced. For a list of supported values, see [Supported languages](#supported-languages).
 
-The text you place for `<title>` shows up as a rollover on the text. The `<pathToFile>`
-is the path to the source file. The `<RegionName>` should be a region in your source
-code that should be included. Use the `#region` and `#endregion` preprocessor syntax
-to specify the region of code to include.
+* `<name>` (*optional*) 
+  * Name for the code snippet. It doesn’t have any impact on the output HTML, but you can use it to improve the readability of your Markdown source. 
 
-For cases where regions don't work, you can specify the start and end of a snippet
-using an XML element name in a single line comment. For example, you could write this in C#:
+* `<pathToFile>` (*mandatory*) 
+  * Relative path in the file system that indicates the code snippet file to reference.
+  
+* `<queryoption>` and `<queryoptionvalue>` (*optional*)
+ * Used together to specify how the code should be retrieved from the file:
+  * `#`:  `#L{startlinenumber}-L{endlinenumber}` (line range) *or* `#{tagname}` (tag name)
+  * `range`: `?range=1,3-5` A range of lines. This example includes lines 1, 3, 4, and 5.
+  * `dedent`: `?dedent=8` Dedents the lines by a number of spaces--in this case, 8. This can be combined with the `range` and other query options that select a subset of the lines of a file.
+  * `outdent`: `?outdent=8` Reverses the indent of the lines by a number of spaces--in this case, 8. This can be combined with `range` and other query options that select a subset of the lines of a file.
 
-```csharp
-// <CodeToInclude>
-int j = 5;
-int i ; 10;
-int sum = i + j;
-// </CodeToInclude>
+We recommend using the tag name option whenever possible. The tag name is the name of a region or of a code comment in the format of `Snippettagname` present in the source code. The following example shows how to refer to the tag name `1`:
+```
+[!code-cs[csrefKeyword#1](../../../../samples/snippets/csharp/language-reference/keywords/throw/throw-1.cs#1)]
 ```
+And you can see how the snippet tags are structured in the [source file](https://github.com/dotnet/docs/blob/master/samples/snippets/csharp/language-reference/keywords/throw/throw-1.cs). For details about tag name representation in code snippet source files by language, see the [DocFX guidelines](https://dotnet.github.io/docfx/spec/docfx_flavored_markdown.html#tag-name-representation-in-code-snippet-source-file).
```

Comment:
I didn't get that the source file linked to a source file. It seems obvious, but... Perhaps change it to "\[this source file](https://github.com/dotnet/docs/blob/master/samples/snippets/csharp/language-reference/keywords/throw/throw-1.cs).")